polygon/bor/bor.go

```diff
@@ -1541,6 +1509,18 @@ func (c *Bor) CommitStates(
 
 	events := chain.Chain.BorEventsByBlock(header.Hash(), blockNum)
 
+	// set this true to check the current stored events vs the current heimdall contents
+	const checkEvents = true
+
+	if checkEvents {
+		if err := heimdall.RemoteEventCheckForBlock(
```

Comment:
Btw this check won't be run for Astrid standalone because the code won't reach here - there is an early `return nil` in the `if c.bridgeReader != nil` statement above which will be used for when running as Astrid standalone. Were you intending on running this check for that too?